### DISCUSSION TITLE
Running 'app:created' nuxt hook before middleware

### DISCUSSION THREAD

**Discussion by hipkiss1991** [nuxt/nuxt#27023] [Questions]
Versions:
```
"nuxt": "^3.7.4",
"@pinia/nuxt": "^0.4.1",
```

Situation:
The project I'm working on is using `nuxtApp.hook('app:created'` in a plugin to setup / initialise the pinia stores / initial data. This includes pinia store setup and a couple of API calls for user data.
When doing so, I am seeing that the middleware `always` runs before this lifecycle hook.
This is not a global middleware.

Question
How do I get the middleware to run `after` the plugin hook has finished.

Tried
- Stand alone middleware file
- page level inline middleware (defining the middleware function here instead of just referencing the file)
- tried defining the middleware in the plugin file, along side the nuxt hook
- Used the `nuxtApp.$pinia` pass through to try and force the correct pinia store state.
- I've tried adding client, hydration, rendering checks in the middleware but it still runs first.
- Combinations of all the above
- Hook plugin is named as `01.setup.js` so its the first file in the plugins dir
- middleware is named as `zzz.js` so its the last file in the middleware dir
- Moved the store init and data fetches to just the plugin itself, instead of inside the hook and still didnt work.

Potential solutions
- I could add the necessary API calls to the middleware but I think this is a really bad idea. This kind of data should only be initialised once, not every transition to this particular path.

```01.setup.js
export default defineNuxtPlugin((nuxtApp) => {
  const authStore = useAuthStore();

  nuxtApp.hook('app:created', async() => {
    await authStore.init();
    await authStore.getUser();
  });
});
```
```myAuth.js
export default defineNuxtRouteMiddleware(async(to, from) => {
  const nuxtApp = useNuxtApp();
  const authStore = await useAuthStore(nuxtApp.$pinia);
  const {
    isUserLogged,
  } = storeToRefs(authStore);

  if (!isUserLogged.value) {
    return navigateTo('/');
  }
});
```

**Comment by hayk98-hayrapetyan** [comment]
just remove the nuxtApp.hook('app:created', async() => { ... }) and immediatly call inside the plugin

```
export default defineNuxtPlugin((nuxtApp) => {
  const authStore = useAuthStore();

  await authStore.init();
  await authStore.getUser();
});
```
